### PR TITLE
Adding new Masthead documentation

### PR DESCRIPTION
**Pull request by beanh66**
Adding a standalone section for masthead documentation and referencing this pattern within the horizontal and vertical navigation patterns.

### CODE REVIEW

File: pattern-library/application-framework/masthead/masthead_overview.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+There are two options for the masthead design, one short option to save space and one tall option to accommodate for larger product logos. The order of information remains consistent regardless of whether you are using the tall or the short masthead design.
+
+## Short Option
+![Image of short masthead](img/masthead_02.png)
+
+## Tall Option
+![Image of tall masthead](img/masthead_01.png)
+
+For [Horizontal Navigation](https://www.patternfly.org/pattern-library/navigation/horizontal-navigation/#_), we recommend the shorter design to reduce vertical real estate, accommodating for the navigation. However, if additional rebranding capabilities are desired, there is an option for a taller masthead which will include a product logo.
```
Comment (andresgalante) [inline pattern-library/application-framework/masthead/masthead_overview.md]: @kybaker Should we allow horizontal nav with tall masthead?

Comment (serenamarie125) [inline pattern-library/application-framework/masthead/masthead_overview.md]: @andresgalante this is a requirement due to rebranding, and needs to continue to be supported.

**Comment by beanh66** [comment]
@andresgalante @cshinn @serenamarie125  Please take a look when you get a chance! I updated the documentation based on your comments!

File: pattern-library/application-framework/masthead/masthead_design.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Masthead
+
+## Design
```
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: No need for this heading

File: pattern-library/application-framework/masthead/masthead_design.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+While there are two options for the masthead design, the order of information remains consistent regardless of whether you are using the tall or the short masthead design.
+
+![Image of masthead details](img/masthead_details.png)
+
+The masthead contains the following items from left to right:
+  1. **Hamburger Menu** The hamburger icon acts as a way for the user to open and close the vertical navigation panel. When using vertical navigation, we recommend always having the hamburger icon visible. When using horizontal navigation, the hamburger menu is only required on mobile devices.
```
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: There should be a ":" after each list item.

File: pattern-library/application-framework/masthead/masthead_design.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    * Any additional icons should be placed to the left of the user, help, and notifications icons.
+  5. **Notifications Icon (optional):**
+    * If notifications or messages are being used, the bell icon should be placed to the left of the help icon. The bell icon is used for opening and closing the notification drawer. Jump to the [Notification Drawer](https://www.patternfly.org/pattern-library/communication/notification-drawer/#/api) pattern for additional details.
+  6. **Help Icon:**
+    * The help icon should always be placed directly to the left of the user icon.
+    * The dropdown arrow should appear to the right of the icon to indicate additional options below.
+    * The dropdown items may vary, but any product documentation or helpful website links available to users should be listed here. This would also be the appropriate location for customer support links.
+    * The dropdown should always include an "About" option that will launch the about modal for product version information. Jump to the "About" pattern for details.
```
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: Should be called "About Modal" and should link to the pattern: http://www.patternfly.org/pattern-library/communication/about-modal/#/api

File: pattern-library/application-framework/masthead/masthead_design.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  2. **Product Logo (optional)** When using the tall masthead design, there is space for a product logo to be placed to the left of the product title for additional branding.
+  3. **Title of Product** The logo area is in the left corner and contains the application product name image file, in the form of a sized SVG graphic.
+  4. **Product Specific Icons (optional):**
+    * Some product may have additional requirements for the masthead. For example, a shopping cart or a mechanism to switch/ launch additional applications.  
+    * Any additional icons should be placed to the left of the user, help, and notifications icons.
+  5. **Notifications Icon (optional):**
+    * If notifications or messages are being used, the bell icon should be placed to the left of the help icon. The bell icon is used for opening and closing the notification drawer. Jump to the [Notification Drawer](https://www.patternfly.org/pattern-library/communication/notification-drawer/#/api) pattern for additional details.
+  6. **Help Icon:**
```
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: For the notification drawer you specific the "bell" icon. You are not specifying the (?) icon here. Is that the proposal? If so, include that note.

File: pattern-library/application-framework/masthead/masthead_design.md (hunk 2 above)
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: No need for this header

File: pattern-library/application-framework/masthead/masthead_design.md (hunk 3 above)
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: Each item should have ":" after it.

File: pattern-library/application-framework/masthead/masthead_design.md (hunk 5 above)
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: You specify to use the bell icon for notifications however do not specify the (?) icon for help. Is that the proposal? If so, say it.

File: pattern-library/application-framework/masthead/masthead_design.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+# Masthead
+
+While there are two options for the masthead design, the order of information remains consistent regardless of whether you are using the tall or the short masthead design.
+
+![Image of masthead details](img/masthead_details.png)
+
+The masthead contains the following items from left to right:
+  1. **Hamburger Menu:** The hamburger icon acts as a way for the user to open and close the vertical navigation panel. When using vertical navigation, we recommend always having the hamburger icon visible. When using horizontal navigation, the hamburger menu is only required on mobile devices.
+  2. **Product Logo (optional):** When using the tall masthead design, there is space for a product logo to be placed to the left of the product title for additional branding.
+  3. **Title of Product:** The logo area is in the left corner and contains the application product name image file, in the form of a sized SVG graphic.
```
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: Sorry, caught this after the fact... to be more inclusive of products and projects, let's call this "Application Title" vs "Title of Product"

File: pattern-library/application-framework/masthead/masthead_design.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Masthead
+
+While there are two options for the masthead design, the order of information remains consistent regardless of whether you are using the tall or the short masthead design.
+
+![Image of masthead details](img/masthead_details.png)
+
+The masthead contains the following items from left to right:
+  1. **Hamburger Menu:** The hamburger icon acts as a way for the user to open and close the vertical navigation panel. When using vertical navigation, we recommend always having the hamburger icon visible. When using horizontal navigation, the hamburger menu is only required on mobile devices.
+  2. **Product Logo (optional):** When using the tall masthead design, there is space for a product logo to be placed to the left of the product title for additional branding.
```
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: To be more inclusive of projects, let's just call this "Logo"

**Comment by LHinson** [comment]
dang, sorry I didn't catch these before the latest PR @beanh66! These are the only two comments I have. Great job on this.

**Comment by mcarrano** [comment]
I think this looks great @beanh66 I just have a general question about visual treatment of masthead items.  In looking at current examples on PatternFly, I see different appearance when the item is selected.  See http://www.patternfly.org/pattern-library/navigation/horizontal-navigation/horizontal-navigation.html#  vs.  http://www.patternfly.org/pattern-library/navigation/vertical-navigation/vertical-navigation.html#_  for example.  Thoughts?

**Comment by beanh66** [comment]
@mcarrano I'm not positive if I'm following, but do you mean the visual appearance when hovering over the masthead options?

**Comment by mcarrano** [comment]
Less about hovering and more about what this looks like when the item is selected and the menu is open.  You'll see in the examples that even the appearance of the menu/pop-up is different between items.  This is probably more of a visual design issue than interaction.

**Comment by andresgalante** [comment]
👍  thanks a lot @beanh66 !

File: pattern-library/application-framework/masthead/masthead_design.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+# Masthead
+
+While there are two options for the masthead design, the order of information remains consistent regardless of whether you are using the tall or the short masthead design.
+
+![Image of masthead details](img/masthead_details.png)
+
+The masthead contains the following items from left to right:
+  1. **Hamburger Menu:** The hamburger icon acts as a way for the user to open and close the vertical navigation panel. When using vertical navigation, we recommend always having the hamburger icon visible. When using horizontal navigation, the hamburger menu is only required on mobile devices.
+  2. **Logo (optional):** When using the tall masthead design, there is space for a product logo to be placed to the left of the product title for additional branding.
```
Comment (LHinson) [inline pattern-library/application-framework/masthead/masthead_design.md]: Formatting should be: **Logo** (optional):

**Comment by LHinson** [comment]
@mcarrano @kybaker @beanh66 per your conversation regarding visuals, what are next steps here?

**Comment by beanh66** [comment]
@LHinson What I have in the masthead dropdown works, I just added a line about the visual treatment and Kyle confirmed that the following example is outdated and should come off the PF site: http://www.patternfly.org/pattern-library/navigation/horizontal-navigation/horizontal-navigation.html#

File: pattern-library/application-framework/masthead/masthead_design.md (hunk 8 above)
Comment (beanh66) [inline pattern-library/application-framework/masthead/masthead_design.md]: @LHinson Thanks, I made the change everywhere!

**Comment by LHinson** [comment]
thanks @beanh66! I've created an issue for the outdated example here: https://github.com/patternfly/patternfly-design/issues/73

I'll go ahead and merge this though!

**Pull request merged**